### DISCUSSION TITLE
Developer should be able to programmatically set `defaultPaymentMethod` via the Greenfield API

### DISCUSSION THREAD

**Discussion by AryanJ-NYC** [btcpayserver/btcpayserver#2793] [Ideas / Feature Requests]
# Story
As a developer, I want to be able to programmatically set the `defaultPaymentMethod` (pictured below) using the `CreateInvoice` function in the Greenfield API.

<img width="463" alt="Screen Shot 2021-08-19 at 11 18 24 AM" src="https://user-images.githubusercontent.com/8263430/130096277-e505c973-bdbd-4ac2-a053-5618cfb2872d.png">

**Comment by Kukks** [comment]
ack

**Reply by Kukks** [reply]
I just checked, and it is already available https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Stores_GetStore

Notice the "defaultPaymentMethod: property

**Reply by AryanJ-NYC** [reply]
It's available on the store level but can we change the `defaultPaymentMethod` on a more granular, invoice-level basis?

I'm realizing now that it may be impossible as the `defaultPaymentMethod` property belongs to a store. D'oh.

Edit: Btw, if not so impossible, I'd happy to throw some sats in for a bounty.

**Reply by bolatovumar** [reply]
@AryanJ-NYC it should be possible. We allow setting the payment method on an invoice-level basis (like if you ca specify that an invoice should be BTC on-chain only or LN-only). It seems reasonable that you should be also able to specify the default payment methods in the same manner. I can take a look into this if nobody else gets to it first.

**Reply by bolatovumar** [reply]
Alright, made some progress. Will put up a PR some time over the next week or two.

**Reply by bolatovumar** [reply]
Draft PR up here: https://github.com/btcpayserver/btcpayserver/pull/2815 (I'm having some problems with testing LN locally hence why it's in draft mode. I did test it though and it works, I just haven't done as much as testing as I would like.)

**Reply by AryanJ-NYC** [reply]
Hey @bolatovumar. I just wanted to say I appreciate your work on this. I'd love to throw you some sats when this is pulled into the codebase. Mention me here and we'll get that all squared away! Thanks again!

**Reply by bolatovumar** [reply]
@AryanJ-NYC alright, it's merged now: https://github.com/btcpayserver/btcpayserver/pull/2815

Feel free to donate here: https://donate.umar-bolatov.com/

**Reply by AryanJ-NYC** [reply]
Thanks @bolatovumar. I just tried donating but my lightning node cannot find a route to yours. Do you have inbound liquidity?

**Reply by bolatovumar** [reply]
@AryanJ-NYC sorry for the late response. I have about ~2.5 million sats of inbound liquidity spread over a few channels. If LN option doesn't work for you then there is probably not a route to my node from yours (also I've been having a lot of issues with my LN node lately, it could be due to that, who knows).

**Comment by Kukks** [comment]
I think it makes sense, given that we do this for some of the other
properties too.
